### PR TITLE
Improving the usage of sets and slices

### LINKED ISSUE #45537: replace all map sets with the new set lib
instead of large number of prs, wasing CI resource, we should update call in one pr

I am working on it, will file an update later

### PR DESCRIPTION
**Pull request by wulianglongrd**
Actually, I don't think it's a PR of much value. The motivation for submitting this PR is that I see `contains` methods in many places and I want to clear them. 

At the same time, when considering the judgment of `contains`, the performance of using `set` is better than that of `slice`, so in some places, `slice` has been replaced by `set`.

### CODE REVIEW

**Review by ericvn — APPROVED** [review]

File: istioctl/pkg/describe/describe.go [hunk 1]
```diff
@@ -220,19 +221,19 @@ func extendFQDN(host string) string {
 }
 
 // getDestRuleSubsets gets names of subsets that match any pod labels (also, ones that don't match).
-func getDestRuleSubsets(subsets []*v1alpha3.Subset, podsLabels []klabels.Set) ([]string, []string) {
-	matchingSubsets := make([]string, 0, len(subsets))
-	nonmatchingSubsets := make([]string, 0, len(subsets))
+func getDestRuleSubsets(subsets []*v1alpha3.Subset, podsLabels []klabels.Set) (sets.String, sets.String) {
```
Comment (howardjohn) [inline istioctl/pkg/describe/describe.go:224]: Why is this changing? It seems worse, now we lose ordering

File: pkg/util/sets/set.go [hunk 2]
```diff
@@ -233,6 +243,11 @@ func (s Set[T]) IsEmpty() bool {
 	return len(s) == 0
 }
 
+// IsNotEmpty indicates whether the set is the not empty set.
+func (s Set[T]) IsNotEmpty() bool {
```
Comment (howardjohn) [inline pkg/util/sets/set.go:247]: nit: I would prefer not to add this, it just gives users two ways to do the same thing `!x.IsEmpty()` or `x.IsNotEmpty()`. Keeping just one ensures consistency

File: pilot/pkg/xds/sds.go [hunk 3]
```diff
@@ -439,30 +439,21 @@ func toEnvoyTLSSecret(name string, certInfo *credscontroller.CertInfo, proxy *mo
 	}
 }
 
-func containsAny(mp sets.Set[model.ConfigKey], keys []model.ConfigKey) bool {
```
Comment (howardjohn) [inline pilot/pkg/xds/sds.go:442]: nit: I don't think everything needs to be a method, if its a niche one-off use case a local function like this is fine.

**Comment by wulianglongrd** [comment]
Actually, I don't think it's a PR of much value. The motivation for submitting this PR is that I see `contains` methods in many places and I want to clear them. At the same time, when considering the judgment of `contains`, the performance of using `set` is better than that of `slice`, so in some places, `slice` has been replaced by `set`.

File: istioctl/pkg/describe/describe.go (hunk 1 above)
Comment (hzxuzhonghu) [reply istioctl/pkg/describe/describe.go:224]: +1

**Review by hzxuzhonghu — CHANGES_REQUESTED** [review]
please give more context what this pr is changing

**Comment by wulianglongrd** [comment]
Comments looks reasonable, close this PR and resubmit #45645 just clean up `contains` func.

**Pull request closed without merging**